tests/core/test_datamodules.py

```diff
@@ -533,6 +534,48 @@ def test_simple_hyperparameters_saving():
     assert data.hparams == AttributeDict({"arg0": 10, "arg1": "foo", "kwarg0": "bar"})
 
 
+def test_define_as_dataclass():
+    class RandomDataset(torch.utils.data.Dataset):
+        def __init__(self, size, length):
+            self.len = length
+            self.data = torch.randn(length, size)
+
+        def __getitem__(self, index):
+            return self.data[index]
+
+        def __len__(self):
+            return self.len
+
+    @dataclass(init=True, repr=True, eq=True, order=True, unsafe_hash=True, frozen=False)
+    class BoringDataModule1(LightningDataModule):
+        def __init__(self, batch_size):
+            super().__init__()
+            self.batch_size = batch_size
```

Comment:
Why not use the dataclass pattern of writing attributes?

```suggestion
        batch_size: int
```

Reply:
This test is to make sure that super().__init__ call can still be made, in case the user wants to explicitly call it with parameters, from __post_init__ in a dataclass initialization. The next test makes use of the dataclass pattern, and doesn't have any __init__ or __post_init__ defined, and doesn't have a super().__init__ call either.